### PR TITLE
[Paddle Inference] General optimization for no_varlen multihead

### PR DESCRIPTION
**Pull request by Wangzheee**
### PR types

Others
### PR changes

Others
### Describe

General optimization for no_varlen multihead: 
- fuse multihead
- reduce data in input tensor(crop the data at the mask)，reduce computation
- use varlen kernel
- qk bias -> mask
- reformat input output for varlen kernel

timeline：0.817ms -> 0.47ms
image
image

### CODE REVIEW

**Review by chenwhql — DISMISSED** [review]
LGTM for PADDLE_ENFORCE

File: paddle/fluid/inference/tensorrt/convert/fc_op.cc [hunk 1]
```diff
@@ -146,6 +146,13 @@ class FcOpConverter : public OpConverter {
     // Declare inputs
     auto* X = engine_->GetITensor(op_desc.Input(i_name).front());
     auto x_dim = X->getDimensions();
+
+    std::cout << "fc::::: " << op_desc.Input(i_name).front()
```
Comment (MARD1NO) [inline paddle/fluid/inference/tensorrt/convert/fc_op.cc]: 这部分调试信息没删

Comment (Wangzheee) [reply paddle/fluid/inference/tensorrt/convert/fc_op.cc]: 收到～

**Review by qingqing01 — COMMENTED** [review]
Need to add unit testing

**Review by ceci3 — COMMENTED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**